libfwupdplugin/fu-progress.c

```diff
@@ -0,0 +1,819 @@
+/*
+ * Copyright (C) 2009 Richard Hughes <richard@hughsie.com>
```

Comment:
Back to the future!

Reply:
Yep! I actually wrote most of this code for zif, a loooong time ago: https://github.com/hughsie/zif/tree/master/libzif :)